python/cudf/cudf/utils/dtypes.py

```diff
@@ -154,7 +154,11 @@ def is_numerical_dtype(obj):
 
 
 def is_string_dtype(obj):
-    return pd.api.types.is_string_dtype(obj) and not is_categorical_dtype(obj)
+    return (
+        not is_decimal_dtype(obj)
+        and pd.api.types.is_string_dtype(obj)
```

Comment:
This feels somewhat fragile, does it happen for any of our other extension types? I.e. Struct, List, etc. types? Why does `pd.api.types.is_string_dtype` return True for a Decimal dtype?

Reply:
This is absolutely fragile. `pd.api.types.is_string_dtype` returns true for any dtype of kind "O" (object), which is also the kind of ExtensionDtype:
```
>>> pandas.core.dtypes.base.ExtensionDtype().kind
'O'
```
There are a few exceptions built in, specifically pandas `Interval` and `Categorical` types, but otherwise _any_ Extension type defaults to behaving as a string type. However, I'm not sure what the implications of that are for our extension types because I don't know exactly how we use this function currently; a quick grep turns it up in a number of places. I assume that pandas does this for the purpose of `repr`, but I'm not sure if that's true and we definitely seem to use it more broadly.

Reply:
I can try making our other extension types also return `False` here and see what breaks. I'd assume that quite a few things would.

Reply:
With groupby (and elsewhere generally), the intent is not to handle struct/list/decimal types implicitly via `is_string_dtype`. Rather, it's a bug that we don't have a default `else` in this `if..elif..else` block, or that we aren't explicitly handling decimal and struct types.

Reply:
I've modified this to return False for all of our other extension types as well. See the updated issue description for a longer discussion.